src/alchemtest/amber/access.py

```diff
@@ -96,12 +94,58 @@ def load_invalidfiles():
         - 'data' : the example of invalid data files
         - 'DESCR': the full description of the dataset
 
+
+    .. deprecated:: 0.7
+       use :func:`load_testfiles` instead
+
     """
 
-    module_path = dirname(__file__)
-    data = [glob(join(module_path, 'invalidfiles/*.out.bz2'))]
+    import warnings  # NOTE: this import is here and not where it should be just because we'll remove this function in next version
+    warnings.warn(
+        "load_invalidfiles() was deprecated in 0.7.0 and will be removed in the following release."
+        " Use load_testfiles() instead",
+        DeprecationWarning)
+    module_path = Path(__file__).parent
+    data = [[
+        module_path / 'testfiles' / 'no_useful_data.out.tar.bz2',
+        module_path / 'testfiles' / 'no_control_data.out.tar.bz2',
+        module_path / 'testfiles' / 'no_temp0_setted.out.tar.bz2',
+        module_path / 'testfiles' / 'no_free_energy_info.out.tar.bz2',
+        module_path / 'testfiles' / 'no_atomic_section.out.tar.bz2',
+        module_path / 'testfiles' / 'no_results_section.out.tar.bz2']]
+
+    with open(module_path / 'testfiles' / 'descr_invalid.rst') as rst_file:
+        fdescr = rst_file.read()
+
+    return Bunch(data=data,
+                 DESCR=fdescr)
+
+
+def load_testfiles():
+    """Load incomplete or wrongly formatted files to be used to test the AMBER parsers.
+
+    Returns
+    -------
+    data : Bunch
+        Dictionary-like object, the interesting attributes are:
+
+        - 'data' : the data files
+        - 'DESCR': the full description of all the files
+
+    """
+
+    testfiles_path = Path(__file__).parent / 'testfiles'
+
+    data = {}
+    for f in testfiles_path.iterdir():
+        f_path = str(f)
+        suffixes = (".out", ".tar", ".bz2")
+        if f.suffix in suffixes:
+            while f.suffix in suffixes:
+                f = f.with_suffix('')
+            data[f.name] = [f_path]
```

Comment:
A simpler way to generate the key as the filename without any suffixes is
```python
key = f.name.replace("".join(f.suffixes), "")
data[key] = [str(f)]
```
It simply replaces all suffixes with the empty string.

It will work with any suffixes and also if there is no suffix.

Reply:
Following up on @xiki-tempula 's https://github.com/alchemistry/alchemtest/pull/71#issuecomment-1273639362 : The clearest approach is to explicitly set up the dictionary, key by key as this will immediately make clear which file belongs to which key.

```python
data = {
 "key1": path / to/ "key1.out.bz2",
 "key2": path / to / "some_other_file.out",
 ...
}
```

Then it's clear that new files need to be added here but it's also clear which file belongs to which key by directly **looking at the code** as opposed to looking at the files and running the algorithm.

As @xiki-tempula says, this is *a lot* better for long-term maintenance.